Model/Data/Customer.php

```diff
@@ -25,61 +25,46 @@
 
 namespace Alma\MonthlyPayments\Model\Data;
 
-use Alma\MonthlyPayments\Helpers\Functions;
 use Magento\Checkout\Model\Session;
-use \Magento\Customer\Api\Data\CustomerInterface;
-use Magento\Framework\Api\SearchCriteriaBuilder;
-use Magento\Framework\Api\SortOrder;
+use Magento\Customer\Api\Data\CustomerInterface;
 use Magento\Payment\Gateway\Data\AddressAdapterInterface;
-use Magento\Sales\Api\OrderRepositoryInterface;
-use Magento\Sales\Model\Order;
 
 class Customer
 {
-    /**
-     * @var SearchCriteriaBuilder
-     */
-    private $searchCriteriaBuilder;
-    /**
-     * @var OrderRepositoryInterface
-     */
-    private $orderRepository;
+
     /**
      * @var Session
      */
     private $checkoutSession;
 
     /**
      * Customer constructor.
-     * @param OrderRepositoryInterface $orderRepository
-     * @param SearchCriteriaBuilder $searchCriteriaBuilder
      * @param Session $checkoutSession
      */
     public function __construct(
-        OrderRepositoryInterface $orderRepository,
-        SearchCriteriaBuilder $searchCriteriaBuilder,
         Session $checkoutSession
-    )
-    {
-        $this->orderRepository = $orderRepository;
-        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+    ) {
         $this->checkoutSession = $checkoutSession;
     }
 
     /**
      * @param CustomerInterface|null $customer
      * @param AddressAdapterInterface[] $addresses
      * @return array
-     * @throws \Magento\Framework\Exception\InputException
      */
-    public function dataFromCustomer($customer, $addresses)
+    public function dataFromCustomer(?CustomerInterface $customer, array $addresses): array
     {
+        $isB2B = false;
+        if (isset($addresses['billing_address'])) {
+            $isB2B = $this->isB2B($addresses['billing_address']);
```

Comment:
```suggestion
            $isB2B = !empty($addresses['billing_address']->getCompany());
```
I think you can now simply remove this method because you just want to test if `empty($address->getCompany())` and this is tested only once... `isB2B` will be useful in the future if we have more things to test ... but with the variable name ... WDYT ?